src/server/tran_server.cpp

```diff
@@ -725,14 +719,27 @@ tran_server::ps_connector::try_connect_to_all_ps (cubthread::entry &)
 	   * TODO It can be too verbose now since it always complain to fail to connect when a PS has been stopped.
 	   * Later on, this job is going to be triggered by a cluster manager or cub_master when a PS is ready to connect.
 	   */
-	  (void) conn->connect ();
+	  if (conn->connect () == NO_ERROR)
+	    {
+	      newly_connected = true;
+	    }
 	}
 
       if (m_terminate.load ())
 	{
 	  break;
 	}
     }
+
+  if (newly_connected)
```

Comment:
```suggestion
  if (newly_connected && !m_terminate.load ())
```
I'm not sure if this `reset_main_connection ()` is required when `m_terminate` is true, what do you think? 

I don't think there will be any functional issues, but I just wonder if there is a need to change the main connection while disconnecting.

Reply:
You're right. It's been also fixed with the change from the `break` to `return` above.

Actually, there is still room for it to be called with your suggestion and the current change because `m_terminate` can be set after checking the condition. However, I think it is acceptable.

Reply:
OK, I understand.